woocommerce/woocommerce-gutenberg-products-block
# Update translation handling to include core translations

Updates translation handling so that core translations are used if present.

Fixes #1238

This PR undoes the translation generation script added in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/486/files since .org now correctly generates json files. This script is no longer needed.

JSON translation files are based on a relative file path converted to md5, so we can generate the correct path by using the WooCommerce core package path and then append the translation script manually. It's not possible to do this using WP core functions.

To avoid affecting the core package, I've introduced a new filter `woocommerce_blocks_register_script`, and added the function to load the translation files to the main plugin file. This file is not used when ran as a package.

### How to test the changes in this Pull Request:

1. Change language to French.
2. Go to dashboard > updates and ensure language packs are up to date.
3. Go to the editor. See the names of the blocks.
4. Check out this PR. Go to the editor. See the names of the blocks. They should now be translated since they are translated in woo core, but not blocks plugin.



### Changelog

> Allow the feature plugin to use woocommerce core translated strings.
